include/vcpkg/sourceparagraph.h

```diff
@@ -72,7 +72,12 @@ namespace vcpkg
         std::vector<Dependency> dependencies;
         std::vector<DependencyOverride> overrides;
         std::vector<std::string> default_features;
-        std::string license; // SPDX license expression
+
+        // there are two distinct "empty" states here
+        // "user did not provide a license" -> nullopt
+        // "user provided license = null" -> {""}
+        Optional<std::string> license; // SPDX license expression
```

Comment:
I would be happier with a specific type rather than needing such mental gymnastics every time we touch this in the future, but as I indicated in chat before I don't think that's the end of the world.

Reply:
If we want to put license types elsewhere, I'll add a type, but I think for now this is fine.